odoo/addons/base/models/res_partner.py

```diff
@@ -986,6 +986,8 @@ def _compute_display_name(self):
                 name = f"{name} <{partner.email}>"
             if partner._context.get('show_vat') and partner.vat:
                 name = f"{name} ‒ {partner.vat}"
+            if partner._context.get("formatted_display_name") and (partner.parent_id or partner.company_name):
```

Comment:
Wait, what ? Who asked that ?

Reply:
You should probably respect other asked content. Here you basically sit on other context key while it makes no sense, we could ask for both a "formatted" version and showing email or vat. Each case should be able to be formatted, if we want to support that feature.